### ISSUE TITLE
vagrant can't find own libs

### ISSUE THREAD

**Issue by Fuuzetsu** [NixOS/nixpkgs#11701]
I'm trying to deploy to AWS with vagrant but get

```
[nix-shell:~/programming/seagate-images]$ vagrant up --provider=aws
/nix/store/528q4mdjg7fmw9f71cc622czyz9wnbjc-vagrant-1.7.4/opt/vagrant/embedded/gems/gems/nokogiri-1.6.3.1/lib/nokogiri.rb:29:in `require': libexslt.so.0: cannot open shared object file: No such file or directory - /nix/store/528q4mdjg7fmw9f71cc622czyz9wnbjc-vagrant-1.7.4/opt/vagrant/embedded/gems/gems/nokogiri-1.6.3.1/lib/nokogiri/nokogiri.so (LoadError)
	from /nix/store/528q4mdjg7fmw9f71cc622czyz9wnbjc-vagrant-1.7.4/opt/vagrant/embedded/gems/gems/nokogiri-1.6.3.1/lib/nokogiri.rb:29:in `rescue in <top (required)>'
	from /nix/store/528q4mdjg7fmw9f71cc622czyz9wnbjc-vagrant-1.7.4/opt/vagrant/embedded/gems/gems/nokogiri-1.6.3.1/lib/nokogiri.rb:25:in `<top (required)>'
	from /home/shana/.vagrant.d/gems/gems/fog-xml-0.1.2/lib/fog/xml.rb:2:in `require'
	from /home/shana/.vagrant.d/gems/gems/fog-xml-0.1.2/lib/fog/xml.rb:2:in `<top (required)>'
	from /home/shana/.vagrant.d/gems/gems/fog-1.36.0/lib/fog.rb:13:in `require'
	from /home/shana/.vagrant.d/gems/gems/fog-1.36.0/lib/fog.rb:13:in `<top (required)>'
	from /home/shana/.vagrant.d/gems/gems/vagrant-aws-0.6.0/lib/vagrant-aws/action/connect_aws.rb:1:in `require'
	from /home/shana/.vagrant.d/gems/gems/vagrant-aws-0.6.0/lib/vagrant-aws/action/connect_aws.rb:1:in `<top (required)>'
	from /home/shana/.vagrant.d/gems/gems/vagrant-aws-0.6.0/lib/vagrant-aws/action.rb:98:in `block in action_read_state'
	from /home/shana/.vagrant.d/gems/gems/vagrant-aws-0.6.0/lib/vagrant-aws/action.rb:96:in `tap'
	from /home/shana/.vagrant.d/gems/gems/vagrant-aws-0.6.0/lib/vagrant-aws/action.rb:96:in `action_read_state'
	from /home/shana/.vagrant.d/gems/gems/vagrant-aws-0.6.0/lib/vagrant-aws/provider.rb:16:in `action'
	from /nix/store/528q4mdjg7fmw9f71cc622czyz9wnbjc-vagrant-1.7.4/opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/machine.rb:180:in `block in action'
	from /nix/store/528q4mdjg7fmw9f71cc622czyz9wnbjc-vagrant-1.7.4/opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/environment.rb:516:in `lock'
	from /nix/store/528q4mdjg7fmw9f71cc622czyz9wnbjc-vagrant-1.7.4/opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/machine.rb:178:in `call'
	from /nix/store/528q4mdjg7fmw9f71cc622czyz9wnbjc-vagrant-1.7.4/opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/machine.rb:178:in `action'
	from /home/shana/.vagrant.d/gems/gems/vagrant-aws-0.6.0/lib/vagrant-aws/provider.rb:32:in `state'
	from /nix/store/528q4mdjg7fmw9f71cc622czyz9wnbjc-vagrant-1.7.4/opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/machine.rb:490:in `state'
	from /nix/store/528q4mdjg7fmw9f71cc622czyz9wnbjc-vagrant-1.7.4/opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/machine.rb:141:in `initialize'
	from /nix/store/528q4mdjg7fmw9f71cc622czyz9wnbjc-vagrant-1.7.4/opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/vagrantfile.rb:79:in `new'
	from /nix/store/528q4mdjg7fmw9f71cc622czyz9wnbjc-vagrant-1.7.4/opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/vagrantfile.rb:79:in `machine'
	from /nix/store/528q4mdjg7fmw9f71cc622czyz9wnbjc-vagrant-1.7.4/opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/environment.rb:612:in `machine'
	from /nix/store/528q4mdjg7fmw9f71cc622czyz9wnbjc-vagrant-1.7.4/opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/plugin/v2/command.rb:168:in `block in with_target_vms'
	from /nix/store/528q4mdjg7fmw9f71cc622czyz9wnbjc-vagrant-1.7.4/opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/plugin/v2/command.rb:192:in `call'
	from /nix/store/528q4mdjg7fmw9f71cc622czyz9wnbjc-vagrant-1.7.4/opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/plugin/v2/command.rb:192:in `block in with_target_vms'
	from /nix/store/528q4mdjg7fmw9f71cc622czyz9wnbjc-vagrant-1.7.4/opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/plugin/v2/command.rb:174:in `each'
	from /nix/store/528q4mdjg7fmw9f71cc622czyz9wnbjc-vagrant-1.7.4/opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/plugin/v2/command.rb:174:in `with_target_vms'
	from /nix/store/528q4mdjg7fmw9f71cc622czyz9wnbjc-vagrant-1.7.4/opt/vagrant/embedded/gems/gems/vagrant-1.7.4/plugins/commands/up/command.rb:74:in `block in execute'
	from /nix/store/528q4mdjg7fmw9f71cc622czyz9wnbjc-vagrant-1.7.4/opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/environment.rb:277:in `block (2 levels) in batch'
	from /nix/store/528q4mdjg7fmw9f71cc622czyz9wnbjc-vagrant-1.7.4/opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/environment.rb:275:in `tap'
	from /nix/store/528q4mdjg7fmw9f71cc622czyz9wnbjc-vagrant-1.7.4/opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/environment.rb:275:in `block in batch'
	from /nix/store/528q4mdjg7fmw9f71cc622czyz9wnbjc-vagrant-1.7.4/opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/environment.rb:274:in `synchronize'
	from /nix/store/528q4mdjg7fmw9f71cc622czyz9wnbjc-vagrant-1.7.4/opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/environment.rb:274:in `batch'
	from /nix/store/528q4mdjg7fmw9f71cc622czyz9wnbjc-vagrant-1.7.4/opt/vagrant/embedded/gems/gems/vagrant-1.7.4/plugins/commands/up/command.rb:58:in `execute'
	from /nix/store/528q4mdjg7fmw9f71cc622czyz9wnbjc-vagrant-1.7.4/opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/cli.rb:42:in `execute'
	from /nix/store/528q4mdjg7fmw9f71cc622czyz9wnbjc-vagrant-1.7.4/opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/environment.rb:301:in `cli'
	from /nix/store/528q4mdjg7fmw9f71cc622czyz9wnbjc-vagrant-1.7.4/opt/vagrant/embedded/gems/gems/vagrant-1.7.4/bin/vagrant:174:in `<main>'
```

However libexslt lives inside the store path and I can manually set the LD_LIBRARY_PATH to help it out

```
LD_LIBRARY_PATH="/nix/store/528q4mdjg7fmw9f71cc622czyz9wnbjc-vagrant-1.7.4/opt/vagrant/embedded/lib/:$LD_LIBRARY_PATH" vagrant up --provider=aws
```

but it seems like a packaging oversight

**Comment by retrry** [comment]
I think I have same problem here https://github.com/NixOS/nixpkgs/issues/9324

**Comment by Fuuzetsu** [comment]
Fixed by #11766 , @retrry if you could confirm for #9324 too that'd be useful
